### ISSUE TITLE
Fails to delete resources when deletion protection is enabled.

### ISSUE THREAD

**Issue by wakeful** [gruntwork-io/cloud-nuke#806]
Hey,

Currently, `cloud-nuke` fails if a resource it is trying to delete has "**deletion protection**" enabled. This feature is available across various resources such as `DynamoDB`, `RDS`, `load balancers`, and others.

What’s your opinion on **extending the logic** to allow `cloud-nuke` to **disable** "**deletion protection**" and **retry** the **deletion**? Or do you think this flag should be removed outside of the _nuke process_?

**Comment by james03160927** [comment]
This is an interesting suggestion. Typically, users would rely on cloud-nuke to clean up resources left over from testing or those no longer needed, regardless of their settings. Therefore, it makes sense to delete such resources with a retry mechanism, even if deletion protection is enabled.

**Comment by wakeful** [comment]
> Typically, users would rely on cloud-nuke to clean up resources left over from testing or those no longer needed, regardless of their settings. Therefore, it makes sense to delete such resources with a retry mechanism, even if deletion protection is enabled.

Exactly, I have the same feeling. However, since this would be a change in behaviour, we could add an extra option to preserve the current behaviour.

Let me create an initial PoC pull request with the changes, and we can discuss it in more detail.

**Comment by james03160927** [comment]
SGTM. Thanks!

**Comment by wakeful** [comment]
hey @james03160927,

I'm still working on this. I should have an initial version ready next week. Sorry, December is always...

**Comment by james03160927** [comment]
No worries. Thanks for the update @wakeful ! :)

**Comment by james03160927** [comment]
Friendly ping @wakeful on this work.
